
**  I made this diff file to change the name of the log file !
the default name is debug.log -> the new name is debug-$USER.log
** I also changed the path where log file is saved, i changed it to 
/var/log/pms-linux/debug-$USER.log
** Now the default conf file is $HOME/.PMS.conf
** Possibility to specify -c or --conf option in order to choose the 
configuration file to use.

===================================================================
--- a/net/pms/PMS.java
+++ b/net/pms/PMS.java
@@ -286,17 +286,21 @@
 		
 		registry = new WinUtils();
 					
+		// Beginning of changes by the Debian/Ubuntu package maintainer.
+		String username;
+		username = System.getProperty("user.name");
 		File debug = null;
 		try {
-			debug = new File("debug.log"); //$NON-NLS-1$
+			debug = new File("/var/log/pms-linux/debug-"+username+".log"); //$NON-NLS-1$
 			pw = new PrintWriter(new FileWriter(debug)); //$NON-NLS-1$
 		} catch (Throwable e) {
-			PMS.minimal("Error in accessing debug.log..."); //$NON-NLS-1$
+			PMS.minimal("Error in accessing debug-"+username+".log..."); //$NON-NLS-1$
 			pw = null;
 		} finally {
 			if (pw == null) {
-				PMS.minimal("Using temp folder for debug.log..."); //$NON-NLS-1$
+				PMS.minimal("Using temp folder for debug-"+username+".log..."); //$NON-NLS-1$
 				debug = new File(configuration.getTempFolder(), "debug.log"); //$NON-NLS-1$
+		// End of modifications from Debian/Ubuntu package maintainer.
 				pw = new PrintWriter(new FileWriter(debug)); //$NON-NLS-1$
 			}
 		}
@@ -1155,8 +1159,9 @@
 		}
 		return compatiblePlayers;
 	}
-	
+
 	public static void main(String args[]) throws IOException, ConfigurationException {
+		String configuration_filename_forced = ""; //$NON-NLS-1$
 		if (args.length > 0) {
 			for(int a=0;a<args.length;a++) {
 				if (args[a].equals("console")) //$NON-NLS-1$
@@ -1167,6 +1172,18 @@
 					System.setProperty("scrollbars", "true"); //$NON-NLS-1$ //$NON-NLS-2$
 				else if (args[a].equals("noconsole")) //$NON-NLS-1$
 					System.setProperty("noconsole", "true"); //$NON-NLS-1$ //$NON-NLS-2$
+				// Possibility to use a desired conf file.
+				else if (args[a].equals("-c") || args[a].equals("--conf")){
+					//PMS.minimal("Using an alternative configuration file ...");
+					if ((a+1) != args.length && args[a+1] != null){
+						configuration_filename_forced = args[a+1]; //$NON-NLS-1$
+						a++;
+					}    
+					else{
+						//PMS.minimal("You specified '" + args[a] + "' option, but you did not set a valid configuration file !!");
+						System.err.println("You specified '" + args[a] + "' option, but you did not set a valid configuration file !!");
+					}
+				} // End of conf file.
 			}
 		}
 		try {
@@ -1178,7 +1195,10 @@
 			if (System.getProperty("noconsole") == null) //$NON-NLS-1$
 				System.setProperty("console", "true"); //$NON-NLS-1$ //$NON-NLS-2$
 		}
-		configuration = new PmsConfiguration();
+		if (configuration_filename_forced.length() == 0)
+			configuration = new PmsConfiguration();
+		else
+			configuration = new PmsConfiguration(configuration_filename_forced);
 		PMS.get();
 		try {
 			// let's allow us time to show up serious errors in the GUI before quitting
